### PR TITLE
Remove kubemacpool w/a in functional test

### PR DESCRIPTION
**Pull request by SchSeba**
After the new version of kubemacpool was integrated into the
operator we can now remove the w/a in our functional tests.

Related to https://github.com/kubevirt/cluster-network-addons-operator/pull/185

---
This change is [Reviewable](https://reviewable.io/reviews/kubevirt/cluster-network-addons-operator/186)

### CODE REVIEW

File: test/e2e/workflow/deployment_test.go [hunk 1]
```diff
@@ -215,12 +207,4 @@ func checkConfigChange(components []Component) {
...
-}
-
-// KubeMacPool is known to restart shortly after first started, try to skip this initial restart
-// https://github.com/kubevirt/cluster-network-addons-operator/issues/141
-// TODO: This should be dropped once KubeMacPool fixes the issue
-func ignoreInitialKubeMacPoolRestart() {
-	By("Ignoring initial KubeMacPool restart")
-	time.Sleep(10 * time.Second)
-}
+}
```
Comment (phoracek) [inline test/e2e/workflow/deployment_test.go]: please add a newline

Comment (SchSeba) [reply test/e2e/workflow/deployment_test.go]: done

**Comment by SchSeba** [comment]
@phoracek I removed the ignore restart only from the deployment in the life cycle.

we can not remove it from the upgrade because old versions of kubemacpool still restart

File: test/check/check.go [hunk 2]
```diff
@@ -157,7 +157,10 @@ func CheckForLeftoverObjects(currentVersion string) {
 	secrets := corev1.SecretList{}
 	err = framework.Global.Client.List(context.Background(), &listOptions, &secrets)
 	Expect(err).NotTo(HaveOccurred())
-	Expect(secrets.Items).To(BeEmpty(), "Found leftover objects from the previous operator version")
+	// TODO: remove this after fixing https://github.com/kubevirt/cluster-network-addons-operator/issues/190
+	for _,secret := range secrets.Items {
```
Comment (phoracek) [inline test/check/check.go]: this needs gofmt

Comment (SchSeba) [reply test/check/check.go]: nice catch Thanks!

File: test/check/check.go [hunk 3]
```diff
@@ -157,7 +157,10 @@ func CheckForLeftoverObjects(currentVersion string) {
 	secrets := corev1.SecretList{}
 	err = framework.Global.Client.List(context.Background(), &listOptions, &secrets)
 	Expect(err).NotTo(HaveOccurred())
-	Expect(secrets.Items).To(BeEmpty(), "Found leftover objects from the previous operator version")
+	// TODO: remove this after fixing https://github.com/kubevirt/cluster-network-addons-operator/issues/190
+	for _, secret := range secrets.Items {
+		Expect(secret.Name).ToNot(Equal("kubemacpool-webhook-secret"), "Found leftover objects from the previous operator version")
```
Comment (phoracek) [inline test/check/check.go]: I'm sorry I missed this one the last time. With your change we don't report which object has been unexpectedly found (unlike the original code). Could you fix it please? I say first drop all KubeMacPool objects from the list and then check if empty.

Comment (SchSeba) [reply test/check/check.go]: done

**Pull request merged**